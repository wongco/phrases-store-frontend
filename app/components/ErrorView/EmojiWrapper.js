import styled from 'styled-components';

const EmojiWrapper = styled.div`
  font-size: 150%;
  margin: 0 0.5em;
`;

export default EmojiWrapper;
