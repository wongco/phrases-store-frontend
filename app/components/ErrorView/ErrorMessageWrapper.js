import styled from 'styled-components';

const ErrorMessageWrapper = styled.div`
  margin: 0.5em;
`;

export default ErrorMessageWrapper;
