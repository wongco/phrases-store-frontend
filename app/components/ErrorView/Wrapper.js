import styled from 'styled-components';

const Wrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  margin: 1em 0;
`;

export default Wrapper;
