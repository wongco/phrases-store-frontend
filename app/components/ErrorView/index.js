import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import Wrapper from './Wrapper';
import EmojiWrapper from './EmojiWrapper';
import ErrorMessageWrapper from './ErrorMessageWrapper';

class ErrorView extends PureComponent {
  static Image = props => <EmojiWrapper>{props.emoji}</EmojiWrapper>;

  static Message = props => (
    <ErrorMessageWrapper>{props.children}</ErrorMessageWrapper>
  );

  render() {
    return <Wrapper>{this.props.children}</Wrapper>;
  }
}

ErrorView.propTypes = {
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.array]),
};

export default ErrorView;
