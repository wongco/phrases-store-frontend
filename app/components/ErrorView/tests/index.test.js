import React from 'react';
import { shallow, mount } from 'enzyme';
import { enzymeFind } from 'styled-components/test-utils';
import toJson from 'enzyme-to-json';
import ErrorView from '../index';
import EmojiWrapper from '../EmojiWrapper';
import ErrorMessageWrapper from '../ErrorMessageWrapper';

describe('<ErrorView />', () => {
  it('renders without crashing', () => {
    shallow(<ErrorView />);
  });

  it('matches snapshot for default criteria', () => {
    const wrapper = shallow(<ErrorView />);
    const serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
  });

  it('should render a prop', () => {
    const id = 'testId';
    const wrapper = mount(<ErrorView id={id} />);
    const renderedComponent = wrapper.find(ErrorView);
    expect(renderedComponent.prop('id')).toBe('testId');
  });

  it('should render its children', () => {
    const children = 'Text';
    const wrapper = mount(<ErrorView>{children}</ErrorView>);
    const renderedComponent = wrapper.find(ErrorView);
    expect(renderedComponent.contains(children)).toBe(true);
  });
});

describe('<ErrorView.Image />', () => {
  it('renders without crashing', () => {
    shallow(<ErrorView.Image />);
  });

  it('matches snapshot for default criteria', () => {
    const wrapper = shallow(<ErrorView.Image />);
    const serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
  });

  it('should render a prop', () => {
    const emoji = 'ðŸ›‘';
    const wrapper = mount(<ErrorView.Image emoji={emoji} />);
    const renderedComponent = wrapper.find(ErrorView.Image);
    expect(renderedComponent.prop('emoji')).toBe('ðŸ›‘');
  });

  it('should render an EmojiWrapper', () => {
    const emoji = 'ðŸ›‘';
    const wrapper = mount(<ErrorView.Image emoji={emoji} />);
    const renderedComponent = enzymeFind(wrapper, EmojiWrapper);
    expect(renderedComponent).toHaveLength(1);
    expect(renderedComponent.text()).toBe(emoji);
  });
});

describe('<ErrorView.Message />', () => {
  it('renders without crashing', () => {
    shallow(<ErrorView.Message />);
  });

  it('matches snapshot for default criteria', () => {
    const wrapper = shallow(<ErrorView.Message />);
    const serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
  });

  it('should render child prop', () => {
    const text = 'testId';
    const wrapper = mount(<ErrorView.Message>{text}</ErrorView.Message>);
    const renderedComponent = enzymeFind(wrapper, ErrorMessageWrapper);
    expect(renderedComponent).toHaveLength(1);
    expect(renderedComponent.text()).toBe(text);
  });
});
