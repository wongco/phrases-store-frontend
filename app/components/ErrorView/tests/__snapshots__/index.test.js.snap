// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorView /> matches snapshot for default criteria 1`] = `<Wrapper />`;

exports[`<ErrorView.Image /> matches snapshot for default criteria 1`] = `<EmojiWrapper />`;

exports[`<ErrorView.Message /> matches snapshot for default criteria 1`] = `<ErrorMessageWrapper />`;
